### DISCUSSION TITLE
How to compile Magick.Native-Q8-x64.dll with new linker directives?

### DISCUSSION THREAD

**Discussion by mclouvem** [dlemstra/Magick.NET#546] [General]
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am using the latest version of Magick.NET

### System Configuration

- Magick.NET version: 7.14.5
- Environment (Operating system, version and so on):
  -Windows 10 1803
  -UWP application 
  -Windows SDK 17134

### Question
I am having some issues to certify my app on Microsoft Store due some problems with Magick.NET-Q8-AnyCPU.dll
One possible solution is recompile the DLL with some linker directives: /safeseh /dynamicbase /nxcompat /appcontainer

Please, Could you give me some directions about how to proper recompile  Magick.NET-Q8-AnyCPU.dll ? Where can I find the source code?

Thank you for all your work. The library is really good!

**Comment by dlemstra** [comment]
Both `/dynamicbase` and `/nxcompat` are on by default. It looks like `/safeseh` is only valid for the `x86` build but I could probably enable that. But with the `/appcontainer` the library will be forced to run inside an app container and I am not sure what happens when I enable that and use the library in a desktop application.

You can find the source for that library here: https://github.com/dlemstra/Magick.Native and there are some minimal building instructions in the Building.md file in the root.

**Comment by mclouvem** [comment]
@dlemstra would be possible to you to build a separated version of Magick.Native with appcontainer directive?
I am trying to build it locally on my laptop, but I am getting a lot of error.

Thank you again!

**Comment by dlemstra** [comment]
Not sure if that would solve your issue maybe other things also need to change? Maybe we could do this in steps to test if it works? I could build a version and add a link here and go from there? This will probably take a while but hope that works for you.

**Comment by mclouvem** [comment]
@dlemstra It sounds good to me. I'm not sure if it would work but I think it worth to try. If you post a link where I can download and submit my app for a new certification on the store

**Reply by jjouppi** [reply]
Hey there! Were you able to get your app on the store? If so, would you possibly be able to let me know how you went about it? Currently running into the same issues as you while using the Nuget package.

Thanks!

**Reply by mclouvem** [reply]
Hi, unfortunately I couldn't submit my app to the store as I couldn't resolve the issue.

**Comment by dlemstra** [comment]
Sorry for taking this long but I finally remember this. I created an x64 version of Magick.native that is compiled with the flags `/dynamicbase /nxcompat /appcontainer` and you can download it here: https://www.dropbox.com/s/sjkoitn8t8um5kl/Magick.Native-Q8-x64.dll?dl=0. This does mean that you will need to switch from the `AnyCPU` version to the `x64` version. And feel free to reach out by email if you want some help to build Magick.Native yourself.

**Comment by nixcamic** [comment]
I don't suppose there would be any chance of having a NuGet packing for the store compatible version? It sure would make things convenient. I noticed there was also interest for it on #186

**Reply by dlemstra** [reply]
If I had clear instructions on what I need to do and how to test for compatibility this I might be able to do that @nixcamic but I don't have enough information to do that now.
